agent/api/task_windows.go

```diff
@@ -50,3 +53,14 @@ func (task *Task) downcaseAllVolumePaths() {
 func getCanonicalPath(path string) string {
 	return filepath.Clean(strings.ToLower(path))
 }
+
+// platformHostConfigOverride Sets up default HostConfig options to be passed to Docker API.
+// Docker for Windows treats some options different than Unix and some are not supported at all.
+// Version 1.12.x of Docker for Windows would ignore the unsupported option MemorySwappiness.
+// Version 17.03.x will cause an error if any value other than -1 is passed in for MemorySwappiness.
+// This bug is not noticed when no value is passed in. However, the go-dockerclient client version
+// we are using removed the json option omitempty causing this parameter to default to 0 if empty.
+// https://github.com/fsouza/go-dockerclient/commit/72342f96fabfa614a94b6ca57d987eccb8a836bf
+func (task *Task) platformHostConfigOverride(hostConfig *docker.HostConfig) {
```

Comment:
nit: I'd probably nest this just a little bit deeper so that the comment you have can be on the part that actually talks about what it does.  Something like this:

```go
// platformHostConfigOverrides applies platform-specific overrides to HostConfig options.
func (task *Task) platformHostConfigOverride(hostConfig *docker.HostConfig) {
	task.forceDefaultMemorySwappiness(hostConfig)
}

// forceDefaultMemorySwappiness sets the MemorySwappiness field to be -1
// Docker for Windows treats some options differently than Linux [...]
func (task *Task) forceDefaultMemorySwappiness(hostConfig *docker.HostConfig) {
	hostConfig.MemorySwappiness = memorySwappinessDefault
}